### PR TITLE
Bach: Create Sample on filter

### PR DESCRIPTION
**Pull request by jansentom**
Allow creating a sample using a custom filter vs. Bernoulli sample.
```
bt_sample = bt.get_sample(table_name='test_data_sample', filter=bt.skating_order % 2 == 0, overwrite=True)
# do super fast stuff
bt_full = bt.get_unsampled()
bt_full.head(10)  # run on ful dataset, epic.
```

### CODE REVIEW

File: bach/bach/dataframe.py [hunk 1]
```diff
@@ -396,40 +397,53 @@ def get_sample(self,
...
+            if sample_percentage:
+                sql = f'''
+                    create temporary table tmp_table_name on commit drop as
+                    ({df.view_sql()});
+                    create table {table_name} as
+                    (select * from tmp_table_name
+                    tablesample bernoulli({sample_percentage}) repeatable ({seed}))
+                '''
+            else:
+                sql = f'create table {table_name} as ({df.view_sql()})'
```
Comment (jansentom) [inline bach/bach/dataframe.py:435]: @thijs-obj can you make this a temp table before merging? That way we can use it in the web demo notebook without too aweful permissions for the user.

Comment (jansentom) [reply bach/bach/dataframe.py]: nevermind, did it already

File: bach/bach/dataframe.py [hunk 2]
```diff
@@ -406,7 +407,7 @@ def get_sample(self,
         """
         original_node = self.get_current_node(name='before_sampling')
         df = self
-        if filter:
+        if filter is not None:
```
Comment (thijs-obj) [inline bach/bach/dataframe.py:410]: not strictly needed, but found it weird to read `if filter` for something where `filter` is sort of a boolean. A boolean could be false, of course a `SeriesBoolean` is actually always True, but still a bit weird to read imho

**Review by thijs-obj — APPROVED** [review]

File: bach/bach/dataframe.py (hunk 2 above)
Comment (jansentom) [reply bach/bach/dataframe.py:410]: makes sense, thanks.

**Pull request merged**